### DISCUSSION TITLE
Support for Android (`libuv.so` access issues)

### DISCUSSION THREAD

**Discussion by sobanieca** [nvim-treesitter/nvim-treesitter#5753] [General]
**Is your feature request related to a problem? Please describe.**

It seems that it's not possible to run `nvim-treesitter` on Android (I'm using Android 14 right now, but in previous version it wasn't possible as well). I'm using Termux with `proot-distro` and `Debian 12` installed and keep getting:

```
Error executing vim.schedule lua callback: Failed to load parser for language 'bash': uv_dlopen: dlopen failed: library "/home/sobanieca/.local/share/nvim/lazy/nvim-treesitter/parser/bash.so" needed or dlopened by "/data/data/com.termux/files/us
r/lib/libuv.so" is not accessible for the namespace "(default)"
stack traceback:
        [C]: in function '_ts_add_language'
        ...s/usr/share/nvim/runtime/lua/vim/treesitter/language.lua:98: in function 'add'
        ...r/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:98: in function 'new'
        ...rmux/files/usr/share/nvim/runtime/lua/vim/treesitter.lua:61: in function '_create_parser'
        ...rmux/files/usr/share/nvim/runtime/lua/vim/treesitter.lua:131: in function 'get_parser'
        ...m/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:141: in function 'ts_highlighter'
        ...m/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:119: in function 'highlighter'
        ...scope.nvim/lua/telescope/previewers/buffer_previewer.lua:248: in function 'cb'
        vim/_editor.lua:263: in function <vim/_editor.lua:262>
```

This is just example for some bash file I've opened, but the issue is for other files as well and it's  always about `libuv.so` not accessible. From what I've read, Android limits access to `libuv.so`, so I believe it won't be possible to use it.
Now I wonder, if there is anything I can do to make it working on Android? I wonder if there are some settings which download TS parses as webassembly? I didn't find such option though.

**Describe the solution you'd like**
Ideally, I would like `nvim-treesitter` or `treesitter` to not have to build parsers in background (if I understand correctly that's what happens), so it won't be necessary to use system wide libraries like `libuv.so`

**Describe alternatives you've considered**
Installing Treesitter CLI on Android, downloading web assembly version of Treesitter.

**Additional context**
Android 14, Samsung Galaxy Fold 5
My treesitter config (I use lazy as plugin manager):

```
{
  "nvim-treesitter/nvim-treesitter",
  build = ":TSUpdate",
  event = "VeryLazy",
  config = function()
    local configs = require("nvim-treesitter.configs"

    configs.setup({
      ensure_installed = {
        "bash",
        "c",
        "css",
        "diff",
        "dockerfile",
        "git_config",
        "git_rebase",
        "gitattributes",
        "gitcommit",
        "gitignore",
        "graphql",
        "hcl",
        "html",
        "http",
        "javascript",
        "jq",
        "jsdoc",
        "json",
        "jsonc",
        "lua",
        "luadoc",
        "luap",
        "make",
        "markdown",
        "markdown_inline",
        "prisma",
        "python",
        "query",
        "regex",
        "sql",
        "terraform",
        "toml",
        "tsx",
        "typescript",
        "vim",
        "vimdoc",
        "xml",
        "yaml",
      },
      sync_install = false,
      highlight = { enable = true },
      indent = { enable = true },
    })
  end
},
```

**Comment by clason** [comment]
I'm sorry, we don't support that platform. Things may work if you set up the environment correctly, but we won't add special code for it.

And `libuv.so` is required for Neovim itself, so this looks like your installation is broken.

(WASM parsers cannot be used from Neovim yet as the tree-sitter C library can't use them -- yet; experimental support was just merged in tree-sitter, but this will take months to trickle down to a Neovim release.)

**Comment by sobanieca** [comment]
Thank you for quick reply. I guess I will have to wait then, maybe something will change in future.
